imports/plugins/core/checkout/client/components/completedOrder.js

```diff
@@ -18,95 +20,228 @@ import AddEmail from "./addEmail";
  * @property {Booleam} isProfilePage - A boolean value that checks if current page is user profile page
  * @return {Node} React node containing the top-level component for displaying the completed order/receipt page
  */
-const CompletedOrder = ({ order, orderId, shops, orderSummary, paymentMethods, handleDisplayMedia, isProfilePage }) => {
-  if (!order) {
-    return (
-      <Components.NotFound
-        i18nKeyTitle="order.notFound"
-        icon="fa fa-barcode"
-        title="Order Not Found"
-      />
-    );
+
+class CompletedOrder extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      order: props.order,
+      orderId: props.orderId,
+      shops: props.shops,
+      orderSummary: props.orderSummary,
+      paymentMethods: props.paymentMethods,
+      handleDisplayMedia: props.handleDisplayMedia,
+      isProfilePage: props.isProfilePage,
+      isCancelled: false
+    };
+    this.cancelOrder = this.cancelOrder.bind(this);
+  }
+  cancelOrder(event) {
```

Comment:
If you write this as an arrow function then you won't need to bind in the constructor. That means, that line 37 won't be needed.
```js
cancelOrder = event => {...}